ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/KeyValueParticipantStateReader.scala

```diff
@@ -8,20 +8,22 @@ import akka.stream.Materializer
 import akka.stream.scaladsl.Source
 import com.daml.ledger.api.health.HealthStatus
 import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlLogEntryId
-import com.daml.ledger.participant.state.kvutils.{Envelope, KVOffset, KeyValueConsumption}
+import com.daml.ledger.participant.state.kvutils.{Envelope, OffsetBuilder, KeyValueConsumption}
 import com.daml.ledger.participant.state.v1._
 import com.daml.lf.data.Time
 import com.daml.metrics.{Metrics, Timed}
 
 class KeyValueParticipantStateReader(reader: LedgerReader, metrics: Metrics)(
     implicit materializer: Materializer)
     extends ReadService {
+  import KeyValueParticipantStateReader.offsetForUpdate
+
   override def getLedgerInitialConditions(): Source[LedgerInitialConditions, NotUsed] =
     Source.single(createLedgerInitialConditions())
 
   override def stateUpdates(beginAfter: Option[Offset]): Source[(Offset, Update), NotUsed] = {
     Source
-      .single(beginAfter.map(KVOffset.onlyKeepHighestIndex))
+      .single(beginAfter)
```

Comment:
This will not work. KeyValueParticipantStateReader acts as a bridge of the two offset notions: the source's notion with a scalar offset and the indexer's that uses the composite offset (a triple of high/middle/low). 

The offset we are given as an argument of this function is a composite, we need to strip the higher elements before passing it on.

Reply:
Yes, indeed -- added more docs around what offsets are expected and changed this specific part to drop only the lowest index (the highest two are expected to be populated by the underlying reader).